modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/portlet/action/EditCPInstanceUnitOfMeasureMVCActionCommand.java

```diff
@@ -0,0 +1,197 @@
+			}
+			else if (cmd.equals(Constants.DELETE)) {
+				_deleteCPInstanceUnitOfMeasures(actionRequest);
+			}
+		}
+		catch (Throwable throwable) {
+			if (throwable instanceof
+					CPInstanceUnitOfMeasureIncrementalOrderQuantityException ||
+				throwable instanceof CPInstanceUnitOfMeasureRateException ||
+				throwable instanceof
+					DuplicateCPInstanceUnitOfMeasureKeyException) {
+
+				SessionErrors.add(
+					actionRequest, throwable.getClass(), throwable);
+
+				hideDefaultErrorMessage(actionRequest);
+
+				String redirect = ParamUtil.getString(
+					actionRequest, "redirect");
+
+				sendRedirect(actionRequest, actionResponse, redirect);
+			}
+			else {
+				_log.error(throwable, throwable);
+
+				throw new PortletException(throwable);
+			}
+		}
+	}
+
+	protected String getSaveAndContinueRedirect(
+			ActionRequest actionRequest,
+			CPInstanceUnitOfMeasure cpInstanceUnitOfMeasure)
+		throws Exception {
+
+		if (cpInstanceUnitOfMeasure == null) {
+			return ParamUtil.getString(actionRequest, "redirect");
+		}
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		return PortletURLBuilder.create(
+			PortletProviderUtil.getPortletURL(
+				actionRequest, themeDisplay.getScopeGroup(),
+				CPDefinition.class.getName(), PortletProvider.Action.EDIT)
+		).setMVCRenderCommandName(
+			"/cp_definitions/edit_cp_instance_unit_of_measure"
+		).setParameter(
+			"cpInstanceId", cpInstanceUnitOfMeasure.getCPInstanceId()
+		).setParameter(
+			"cpInstanceUnitOfMeasureId",
+			cpInstanceUnitOfMeasure.getCPInstanceUnitOfMeasureId()
+		).setWindowState(
+			LiferayWindowState.POP_UP
+		).buildString();
+	}
+
+	private CPInstanceUnitOfMeasure _addOrUpdateCPInstanceUnitOfMeasure(
+			ActionRequest actionRequest)
+		throws Exception {
+
+		long cpInstanceUnitOfMeasureId = ParamUtil.getLong(
+			actionRequest, "cpInstanceUnitOfMeasureId");
+		long cpInstanceId = ParamUtil.getLong(actionRequest, "cpInstanceId");
+		boolean active = ParamUtil.getBoolean(actionRequest, "active");
+		BigDecimal incrementalOrderQuantity = (BigDecimal)ParamUtil.getNumber(
+			actionRequest, "incrementalOrderQuantity", BigDecimal.ZERO);
+		String key = ParamUtil.getString(actionRequest, "key");
+		Map<Locale, String> nameMap = _localization.getLocalizationMap(
+			actionRequest, "name");
+		int precision = ParamUtil.getInteger(actionRequest, "precision");
+		boolean primary = ParamUtil.getBoolean(actionRequest, "primary");
+		double priority = ParamUtil.getDouble(actionRequest, "priority");
+		BigDecimal rate = (BigDecimal)ParamUtil.getNumber(
+			actionRequest, "rate", BigDecimal.ONE);
+		String sku = ParamUtil.getString(actionRequest, "sku");
+
+		if (cpInstanceUnitOfMeasureId > 0) {
+			CPInstanceUnitOfMeasure cpInstanceUnitOfMeasure =
+				_cpInstanceUnitOfMeasureService.fetchCPInstanceUnitOfMeasure(
+					cpInstanceUnitOfMeasureId);
+
+			if (cpInstanceUnitOfMeasure.isPrimary()) {
```

Comment:
this looks wrong to me

Reply:
When a checkbox is disabled and a form submission is made the value that is submitted is false.
Since if a UoM is primary it is not possible to change the value of the checkbox to false we have decided on this saving logic